queries.sql

```diff
@@ -25,3 +25,100 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
 /*find all animals with a weight between 10.4kg and 17.3kg(including 
 the weights that equals precisely 10.4kg or 17.3kg)*/
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+--First Transaction
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+COMMIT;
+SELECT * FROM animals;
+
+--Second Transaction
+BEGIN;
+UPDATE animals
+SET species = ' ';
+SELECT * FROM animals;
+ROLLBACK;
+
+--Transaction
+/* Update the animals table by setting the species column to digimon
+for all animals that have a name ending in mon */
+BEGIN;
+UPDATE animals
+SET species = 'digimon';
+WHERE name LIKE '%mon';
+SELECT * FROM animals
+
+/*Update the animals table by setting the species column
+to pokemon for all animals that do not have species already set*/
+UPDATE animals
+SET species='pokemon'
+WHERE species IS NULL OR TRIM(species)='';
+SELECT count(*) FROM animals WHERE species='pokemon';
+SELECT * FROM animals;
```

Comment:
---

Hello @IvonneBenitesRodriguez,

I've reviewed your submitted code in relation to the project requirements. Here's the feedback:

1. **Syntax Error:** You have a misplaced semicolon in the `UPDATE` statement for setting the species to 'digimon'. The correct syntax should be:
```sql
UPDATE animals
SET species = 'digimon'
WHERE name LIKE '%mon';
```
Note the removal of the semicolon after `'digimon'`.

2. **Transaction Commit Missing:** The project requirements specify that you should commit the transaction after making and verifying the changes. Your current code lacks the `COMMIT;` statement to finalize the changes.

3. **Verification After Commit Missing:** After committing the transaction, there's no subsequent verification to ensure the changes persist. You should have another `SELECT` statement after the `COMMIT;` to meet this requirement.

Here's a suggested structure for your code:

```sql
BEGIN; -- start transaction

-- Update species to 'digimon' for names ending in 'mon'
UPDATE animals
SET species = 'digimon'
WHERE name LIKE '%mon';

-- Update species to 'pokemon' where species is not set
UPDATE animals
SET species = 'pokemon'
WHERE species IS NULL OR TRIM(species) = '';

-- Verify the changes
SELECT * FROM animals;

COMMIT; -- commit the transaction

-- Verify that changes persist after commit
SELECT * FROM animals;

```

Kindly make these adjustments and resubmit your code for further review. Great effort so far, and let me know if you have any questions or need more clarification!

Warm regards,
Temitope

---

Reply:
Dear @topeogunleye,
thanks for your explanations about commit transaction, and the importance about verification.
I modified this part also according your suggestion,
Best regards and thanks again,